mypy/build.py

```diff
@@ -1056,11 +1058,10 @@ def validate_meta(meta: Optional[CacheMeta], id: str, path: Optional[str],
 
     # TODO: Share stat() outcome with find_module()
     path = os.path.abspath(path)
-    # TODO: Don't use isfile() but check st.st_mode
-    if not os.path.isfile(path):
+    st = manager.get_stat(path)  # TODO: Errors
```

Comment:
What about adding a `is_file` method to `BuildManager` instead and using it here?

Reply:
That's a bigger effort, we'd have to give BuildManager a cache results and use that everywhere. There's also an inefficiency in find_module(), which seems to make way to many stat() and listdir() calls, which I haven't tracked down.